ipn/ipnserver/server.go

```diff
@@ -82,66 +81,68 @@ type Server struct {
 	logf         logger.Logf
 	backendLogID string
 	// resetOnZero is whether to call bs.Reset on transition from
-	// 1->0 connections.  That is, this is whether the backend is
+	// 1->0 active HTTP requests. That is, this is whether the backend is
 	// being run in "client mode" that requires an active GUI
-	// connection (such as on Windows by default).  Even if this
+	// connection (such as on Windows by default). Even if this
 	// is true, the ForceDaemon pref can override this.
 	resetOnZero bool
 
 	// mu guards the fields that follow.
 	// lock order: mu, then LocalBackend.mu
 	mu         sync.Mutex
 	lastUserID string // tracks last userid; on change, Reset state for paranoia
-	allClients map[net.Conn]*ipnauth.ConnIdentity
+	activeReqs map[*http.Request]*ipnauth.ConnIdentity
 }
 
 // LocalBackend returns the server's LocalBackend.
 func (s *Server) LocalBackend() *ipnlocal.LocalBackend { return s.b }
 
-// bufferIsConnect reports whether br looks like it's likely an HTTP
-// CONNECT request.
-//
-// Invariant: br has already had at least 4 bytes Peek'ed.
-func bufferIsConnect(br *bufio.Reader) bool {
-	peek, _ := br.Peek(br.Buffered())
-	return mem.HasPrefix(mem.B(peek), mem.S("CONN"))
-}
-
-func (s *Server) serveConn(ctx context.Context, c net.Conn, logf logger.Logf) {
-	// First sniff a few bytes to check its HTTP method.
-	br := bufio.NewReader(c)
-	c.SetReadDeadline(time.Now().Add(30 * time.Second))
-	br.Peek(len("GET / HTTP/1.1\r\n")) // reasonable sniff size to get HTTP method
-	c.SetReadDeadline(time.Time{})
-
-	// Handle logtail CONNECT requests early. (See docs on handleProxyConnectConn)
-	if bufferIsConnect(br) {
-		s.handleProxyConnectConn(ctx, br, c, logf)
+func (s *Server) serveHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "CONNECT" {
+		s.handleProxyConnectConn(w, r)
```

Comment:
add a comment about this only being used by the windows GUI. Probably also build tag it as such?

Reply:
It has a lot of docs on the definition of the func, but I'll add a little here too.